### ISSUE TITLE
SNIMissingWarning when using AWS CLI

### ISSUE THREAD

**Issue by bczoma** [travis-ci/travis-ci#10141]
Noticed that the build is now full of warnings. It doesn't seem to have an effect on the outcome of the test but should be resolved.

I went back to build history and it started around Aug 30, getting a lot of warnings like this:

```
$ aws cloudformation validate-template --template-body file://templates/solace-master.template
/home/travis/.local/lib/python2.7/site-packages/urllib3/util/ssl_.py:369: SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning
```

See build example:
https://travis-ci.org/SolaceDev/solace-aws-ha-quickstart/builds/423035972

A few days before the issue was not there:
https://travis-ci.org/SolaceDev/solace-aws-ha-quickstart/builds/418838756

There is a history of seemingly related Travis issues, not sure what exactly happened this time
https://github.com/travis-ci/travis-ci/issues?utf8=%E2%9C%93&q=is%3Aissue+SNIMissingWarning

**Comment by BanzaiMan** [comment]
It seems to me that the issue has always been there for your repository. See, for example, https://travis-ci.org/SolaceDev/solace-aws-ha-quickstart/builds/369315919#L452

If you look at [the documentation](https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings) that the warning message points to, you will notice this:

>  `SNIMissingWarning`
>      This happens on Python 2 versions older than 2.7.9. These older versions lack SNI support. This can cause servers to present a certificate that the client thinks is invalid. Follow the pyOpenSSL guide to resolve this warning.

So, it seems to me that the warning is for Using an old Python, and is to be expected.

**Comment by bczoma** [comment]
Thank you @BanzaiMan . What is the recommended way to upgrade to python 2.7 latest?

I have tried multiple ways without success... keep getting: 

```
sudo apt-get install python2.7
Reading package lists...
Building dependency tree...
Reading state information...
python2.7 is already the newest version (2.7.6-8ubuntu0.4).
```

**Comment by BanzaiMan** [comment]
Is Ruby a hard requirement for your repository? If not, switching to `language: python` and choosing a more recent version might be your best bet.

**Comment by bczoma** [comment]
Changing to `language: python` has addressed it, now the version used is `Python 2.7.14`.

The ruby support in this distribution is good enough for me:
`ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]`

Thanks for the advise!

This can be closed now.
